### PR TITLE
New pattern for http hook

### PR DESCRIPTION
**Pull request by dmvieira**

### CODE REVIEW

File: barterdude/hooks/metrics/prometheus.py [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
-class Prometheus(BaseHook):
-    def __init__(self, gateway_url: str):
-        self.__gateway_url = gateway_url
+try:
+    from prometheus_client import CollectorRegistry, generate_latest
+except ImportError:
+    print("""
+    Please install extra dependency with:
+        `pip install barterdude[prometheus]`
+    """)
```
Comment (gligneul) [inline barterdude/hooks/metrics/prometheus.py]: Isso é mesmo necessário? Não deveríamos colocar apenas no requiriments.txt?

File: tests/helpers.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        Mock(startup=CoroutineMock(), shutdown=CoroutineMock()),
+    )
+
+
+def get_app():
+    return MyApp(connections=[AMQPConnection(  # nosec
+                name="conn1",
+                hostname="localhost",
+                username="username",
+                password="pass"
```
Comment (gligneul) [inline tests/helpers.py]: Não deveria ser guest/guest?

File: barterdude/hooks/metrics/prometheus.py (hunk 1 above)
Comment (dmvieira) [reply barterdude/hooks/metrics/prometheus.py]: This is for optional install.... There are other ways to measure metric and Prometheus is just one.... This should be on requirements_dev.txt too

File: tests/helpers.py (hunk 2 above)
Comment (dmvieira) [reply tests/helpers.py]: It's just a "mock"... No connection happens

**Pull request merged**